juju/juju
# Feature/hide juju managed units

## Description of change

Hide `juju-managed-units` from user(immutable for users, always equals to `false`)

## QA steps
1. unit tests ensure.
2. deploy any caas charm to caas model with `juju-managed-units=true` configed by `juju deploy <charm-build-dir> --config juju-managed-units=true`, then run `kubectl get deploy -n <caas-model-name>`, the `deployment` resource of the app should be created.

##### deploy with this branch and develop branch build with `juju-managed-units=true`
```bash
juju deploy ./builds/mysql  --config juju-managed-units=true 
```
```bash
kubectl get all -n testcaas1   # k8s managed created svc and deployment resources <this PR>
```
```
NAME                DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deploy/juju-mysql   1         1         1            0           42s

NAME                       DESIRED   CURRENT   READY     AGE
rs/juju-mysql-6fc5f55fbf   1         1         0         42s

NAME                DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deploy/juju-mysql   1         1         1            0           42s

NAME                       DESIRED   CURRENT   READY     AGE
rs/juju-mysql-6fc5f55fbf   1         1         0         42s

NAME                             READY     STATUS              RESTARTS   AGE
po/juju-mysql-6fc5f55fbf-n2gbz   0/1       ContainerCreating   0          42s
po/juju-operator-mysql           1/1       Running             0          2m

NAME             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
svc/juju-mysql   ClusterIP   10.152.183.23   <none>        3306/TCP   42s
```

```bash
kubectl get all -n testcaas2   # juju managed only created svc but no deployment resources <`current develop branch`>
```
```
NAME                     READY     STATUS    RESTARTS   AGE
po/juju-operator-mysql   1/1       Running   0          28s
po/juju-unit-mysql-0     1/1       Running   0          18s

NAME             TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
svc/juju-mysql   ClusterIP   10.152.183.245   <none>        3306/TCP   18s
```


## Documentation changes

Config var description of `juju-managed-units` has been updated 

## Bug reference
it's a feature change but not a bug
